webapp/src/DOMJudgeBundle/Controller/API/JudgehostController.php

```diff
@@ -689,185 +701,91 @@ public function addJudgingRunAction(
+        } else {
+            $required = [
+                'testcaseid',
+                'runresult',
+                'runtime',
+                'output_run',
+                'output_diff',
+                'output_error',
+                'output_system'
+            ];
 
-        foreach ($required as $argument) {
-            if (!$request->request->has($argument)) {
-                throw new BadRequestHttpException(
-                    sprintf("Argument '%s' is mandatory", $argument));
+            foreach ($required as $argument) {
+                if (!$request->request->has($argument)) {
+                    throw new BadRequestHttpException(
+                        sprintf("Argument '%s' is mandatory", $argument));
+                }
             }
-        }
 
-        $testCaseId   = $request->request->get('testcaseid');
-        $runResult    = $request->request->get('runresult');
-        $runTime      = $request->request->get('runtime');
-        $outputRun    = $request->request->get('output_run');
-        $outputDiff   = $request->request->get('output_diff');
-        $outputError  = $request->request->get('output_error');
-        $outputSystem = $request->request->get('output_system');
+            $testCaseId = $request->request->get('testcaseid');
+            $runResult = $request->request->get('runresult');
+            $runTime = $request->request->get('runtime');
+            $outputRun = $request->request->get('output_run');
+            $outputDiff = $request->request->get('output_diff');
+            $outputError = $request->request->get('output_error');
+            $outputSystem = $request->request->get('output_system');
+
+            /** @var Judgehost $judgehost */
+            $judgehost = $this->entityManager->getRepository(Judgehost::class)->find($hostname);
+            if (!$judgehost) {
+                return;
+            }
 
-        /** @var Judgehost $judgehost */
+            /** @var Judging $judging */
+            $judging = $this->entityManager->getRepository(Judging::class)->find($judgingId);
+            $this->addSingleJudgingRun($hostname, $judgingId, $testCaseId, $runResult, $runTime, $judging, $outputSystem, $outputError, $outputDiff, $outputRun);
+            $judgehost->setPolltime(Utils::now());
+            $this->entityManager->flush();
+        }
+    }
+
+    private function addMultipleJudgingRuns(Request $request, string $hostname, int $judgingId) {
+        $judgingRuns = json_decode($request->request->get('batch'), TRUE);
+        if (!is_array($judgingRuns)) {
+            throw new BadRequestHttpException(
+                sprintf("Argument 'batch' is not an array"));
+        }
+        /** @var Judgehost $judgehost */ // <--- FIXME: do this in the other commit
         $judgehost = $this->entityManager->getRepository(Judgehost::class)->find($hostname);
         if (!$judgehost) {
+            // FIXME: silent return??? also at the original place
```

Comment:
The old v4 code did the same, but we might want to throw an exception instead?

Reply:
Done.